cdap-watchdog/src/main/java/co/cask/cdap/metrics/collect/MapReduceCounterCollectionService.java

```diff
@@ -64,6 +69,23 @@ protected void publish(Iterator<MetricValue> metrics) throws Exception {
     }
   }
 
+  public static Map<String, String> parseTags(String counterGroupName) {
+    // see publish method for format info
+
+    if (counterGroupName.startsWith("cdap.")) {
```

Comment:
it's more like an assertion. If it doesn't start from "cdap." then we cannot get any information about the context. I think I can even turn it into check+throw..